var/spack/repos/builtin/packages/python/package.py

```diff
@@ -303,6 +305,59 @@ def determine_variants(cls, exes, version_str):
             except ProcessError:
                 variants += '~tix'
 
+        # Get config args for shared, debug, and optimizations variant checks
+        sysconf_imp = 'import sysconfig'
+        if version < Version('2.7'):
+            sysconf_imp = 'import distutils.sysconfig as sysconfig'
+        conf_args = python(
+            '-c',
+            '%s; print(sysconfig.get_config_var("CONFIG_ARGS"))' % sysconf_imp,
+            output=str, error=str)
+
+        # check for shared variant
+        # defualt is on, so we check for the string to turn it off
```

Comment:
The default for this variant in this package is on, but `./configure --help` in the cpython repo produces:
```
  --enable-shared         enable building a shared Python library (default is
                          no)
```

I think the right way to do these checks is to comb through all the variables and *avoid* using `CONFIG_ARGS`. For this one, it would be `sysconfig.get_config_var('Py_ENABLE_SHARED')`, which returns 1 or 0.

Reply:
It looks like `RUNSHARED` is stable across py2 and py3, and `sysconfig.get_config_var('RUNSHARED')` will return a non-empty string if the python executable has a shared library. Note that the error message from the clingo cmake build if you provide a python executable *without* a shared library is very difficult to parse -- it just says `Python_LIBRARIES` was not found, even if you define it on the same command line.

Reply:
@cosmicexplorer I've pushed an improved version of this, let me know what you think.